File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> result = new ArrayList<>();
        Map<String, List<String>> map = new HashMap<>();
        int i = 0;
        while(i < strs.length){
            char[] charArray = strs[i].toCharArray();
            Arrays.sort(charArray);
            String str = new String(charArray);
            if(map.containsKey(str)){
                map.get(str).add(strs[i]);
            }
            else{
                map.put(str, new ArrayList<>());
                map.get(str).add(strs[i]);
            }
            i++;
        }
        for(Map.Entry<String, List<String>> curr : map.entrySet()){
            result.add(curr.getValue());
        }
        return result;
        
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}
        for strings in strs:
            sorted_string = ''.join(sorted(strings))
            if sorted_string in hashmap:
                hashmap.get(sorted_string).append(strings)
            else:
                hashmap[sorted_string] = []
                hashmap.get(sorted_string).append(strings)
        return list(hashmap.values())

      



        